src/context.ts

```diff
@@ -5,6 +5,15 @@ import { QueryResult } from './graphql'
 import * as GQL from './schema/graphqlschema'
 import { ConfigurationCascadeOrError, ConfigurationSubject, ID, Settings } from './settings'
 
+export type UpdateExtensionSettingsArgs =
+    | { edit?: ConfigurationUpdateParams }
+    | {
+          extensionID: string
+          // TODO: unclean api, allows 4 states (2 bools), but only 3 are valid (none/disabled/enabled)
```

Comment:
what prevents you from making this something like `state: 'none' | 'disabled' | 'enabled'`? Or simply `enabled?: boolean`

Reply:
That would be a nice improvement, but I'm focused on building the message passing right now